File: python/current-arrest.py
#!/usr/bin/env python

# Written for Python 2.x
# Tested on
# Windows 7 x64 - Python 2.7.6
# Ubuntu 14.04 LTS (trusty) - Python 2.7.6
# OS X 10.9.5 - Python 2.7.5
# OS X 10.10.2 - Python 2.7.6

import urllib
import json

if __name__ == "__main__":
  # get the arrest data
  arrests = urllib.urlopen("http://api.canyonco.org/Sheriff/CurrentArrest").read()
  # parse the JSON
  jsonData = json.loads(arrests)

  # If the first element's Error is not empty it will be the error message
  # not exactly restful, but it was easy
  error = jsonData[0]["Error"]

  if error == '':
    # loop through the inmates
    for inmate in jsonData:
      print "%s, %s" % (inmate["LastName"].title(), inmate["FirstName"].title())

      # loop through the arrests
      for arrest in inmate["Arrests"]:
        print "  Arrest Date: %s" % arrest["ArrestDate"]

      # loop through the charges
      for charge in inmate["Charges"]:
        print "  Charge: %s: %s" % (charge["StatuteCode"], charge["StatuteDesc"])

  else:
    print "Error getting data", error

File: perl/current-arrest.pl
#!/usr/bin/env perl

# Written for perl 5.x
# My perl is five years rusty - apologies to Monks out there
# Tested on
# Ubuntu 14.04 LTS (trusty) - perl (v5.18.2)

use strict;
use warnings;

use JSON;        # cpan JSON
use LWP::Simple; #cpan LWP::Simple

# get the arrest data
my $data = get("http://api.canyonco.org/Sheriff/CurrentArrest");
die "Couldn't get it!" unless defined $data;

# parse the JSON
my $json = JSON->new->allow_nonref;
my $inmates = $json->decode($data);
my $error = "The error was...";

# loop through the inmates
foreach my $inmate(@{$inmates}) {
  
    # If the first element's Error is not empty it will be the error message
    # not exactly restful, but it was easy
  $error = $inmate->{Error}; 
  last if $error ne "";
  
  print "$inmate->{LastName}, $inmate->{FirstName} $inmate->{MiddleName}\n";

  # loop through the arrests
  foreach my $arrest(@{$inmate->{Arrests}}) {
    print "  Arrest Date: $arrest->{ArrestDate}\n";
  }

  # loop through the charges
  foreach my $charge(@{$inmate->{Charges}}) {
    print "  Charge: $charge->{StatuteCode}: $charge->{StatuteDesc}\n";
  }
}

# if error was set - display it
print "Error: $error\n" if $error ne "";
